src/main/java/com/powsybl/openloadflow/util/PropagatedContingency.java

```diff
@@ -46,55 +49,88 @@ public int getIndex() {
+
+        for (Switch sw : switchesToOpen) {
+            branchIdsToOpen.add(sw.getId());
+        }
+
+        for (Terminal terminal : terminalsToDisconnect) {
+            if (terminal.getConnectable() instanceof Branch) {
+                branchIdsToOpen.add(terminal.getConnectable().getId());
+            }
+        }
     }
 
-    public static List<PropagatedContingency> create(Network network, List<Contingency> contingencies, Set<Switch> allSwitchesToOpen) {
+    public static List<PropagatedContingency> createListForSensitivityAnalysis(Network network, List<Contingency> contingencies) {
         List<PropagatedContingency> propagatedContingencies = new ArrayList<>();
         for (int index = 0; index < contingencies.size(); index++) {
             Contingency contingency = contingencies.get(index);
-            PropagatedContingency propagatedContingency = new PropagatedContingency(contingency, index);
-            propagatedContingencies.add(propagatedContingency);
-
-            Set<Switch> switchesToOpen = new HashSet<>();
-            Set<Terminal> terminalsToDisconnect =  new HashSet<>();
-            for (ContingencyElement element : contingency.getElements()) {
-                switch (element.getType()) {
-                    case BRANCH:
-                    case LINE:
-                    case TWO_WINDINGS_TRANSFORMER:
-                        // branch check is done inside branch tripping
-                        new BranchTripping(element.getId(), null)
-                            .traverse(network, null, switchesToOpen, terminalsToDisconnect);
-                        propagatedContingency.getBranchIdsToOpen().add(element.getId());
-                        break;
-                    case HVDC_LINE:
-                        HvdcLine hvdcLine = network.getHvdcLine(element.getId());
-                        if (hvdcLine == null) {
-                            throw new PowsyblException("HVDC line '" + element.getId() + "' not found");
-                        }
-                        propagatedContingency.getHvdcIdsToOpen().add(element.getId());
-                        break;
-                    default:
-                        //TODO: support all kinds of contingencies
-                        throw new UnsupportedOperationException("TODO");
-                }
+            PropagatedContingency propagatedContingency = PropagatedContingency.create(network, contingency, index);
+            Optional<Switch> coupler = propagatedContingency.switchesToOpen.stream().filter(PropagatedContingency::isCoupler).findFirst();
+            if (coupler.isEmpty()) {
+                propagatedContingencies.add(propagatedContingency);
+            } else {
+                // Sensitivity analysis works in bus view, it cannot deal (yet)  with contingencies whose propagation encounters a coupler
+                LOGGER.warn("Propagated contingency '{}' not processed: coupler '{}' has been encountered while propagating the contingency",
```

Comment:
I don't know if coupler has a meaning in English indeed. But for sure you have to precise that it is between two bus bar sections.

Reply:
Coupler is a well known term in power system

Reply:
Here you skip the contingency, so it means that the result will not contain this contingency. The other solution is to set a predefined result. What is preferable?